src/resources/parser/texture/ktx2.js

```diff
@@ -0,0 +1,202 @@
+import { http } from '../../../net/http.js';
+import { basisTranscode } from '../../basis.js';
+import { Texture } from '../../../graphics/texture.js';
+import { ADDRESS_CLAMP_TO_EDGE, ADDRESS_REPEAT, TEXHINT_ASSET } from '../../../graphics/constants.js';
+
+// helper class for organized reading of memory
+class ReadStream {
```

Comment:
Should this go in the core folder? Any scope for utilizing in any other modules?

Reply:
Yes I made a similar class for hdr loading and would like to extract this and have both use the same interface. Just didn't want to overcomplicate this PR.